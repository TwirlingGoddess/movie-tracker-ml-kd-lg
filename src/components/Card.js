import React from 'react';
import PropTypes from 'prop-types'

export const Card = ({title, overview, rating}) => {
  return(
    <div className="Card">
      <h2>{title}</h2>
      <p>{overview}</p>
      <h3>{rating}</h3>
    </div>
    )
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired
}