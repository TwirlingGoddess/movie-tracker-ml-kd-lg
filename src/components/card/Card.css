.card {
  width: 33%; }

.fave-clicked {
  background-color: #da6039;
  outline: none; }
