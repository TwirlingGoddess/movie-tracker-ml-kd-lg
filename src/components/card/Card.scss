.card {
  width: 33%;
}

.fave-clicked {
  background-color: #da6039;
}