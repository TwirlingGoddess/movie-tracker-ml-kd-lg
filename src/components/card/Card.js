import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

export const Card = ({ title, overview, rating, image }) => {
  return (
    <div className="card">
      <h2>{title}</h2>
      <img src={image} alt="movie image" />
      <h3>Rating: {rating}</h3>
      <p>{overview}</p>
      <button>FAVORITE</button>
    </div>
  );
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired
};
