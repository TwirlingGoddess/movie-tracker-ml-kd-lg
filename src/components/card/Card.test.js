import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let mockSaveFavorite = jest.fn();
  let mockEvaluateClass = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Card saveFavorite={mockSaveFavorite} evaluateClass={mockEvaluateClass} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('saveFavorites should be invoked on when favorite button is clicked', () => {
    wrapper.find('button').simulate('click');

    expect(mockSaveFavorite).toHaveBeenCalled();
  });
});
