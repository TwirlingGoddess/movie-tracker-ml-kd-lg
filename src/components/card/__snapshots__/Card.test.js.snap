// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    evaluateClass={[Function]}
    saveFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <img
          alt="movie poster"
          src={undefined}
/>,
        <h3>
          Rating: 
</h3>,
        <p />,
        <p />,
        <button
          className={undefined}
          onClick={[Function]}
>
          FAVORITE
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Rating: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Rating: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FAVORITE",
          "className": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "FAVORITE",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <img
            alt="movie poster"
            src={undefined}
/>,
          <h3>
            Rating: 
</h3>,
          <p />,
          <p />,
          <button
            className={undefined}
            onClick={[Function]}
>
            FAVORITE
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Rating: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Rating: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FAVORITE",
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "FAVORITE",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
